src/pybind/rbd/rbd.pyx

```diff
@@ -4463,6 +4463,83 @@ written." % (self.name, ret, length))
         """
         return ConfigImageIterator(self)
 
+
+    def config_image_set(self, key, value):
+        """
+        Set an image-level configuration override.
+
+        :param key: config key
+        :type key: str
+        :param value: config value
+        :type value: str
+        """
+        key = 'conf_' + key
+        key = cstr(key, 'key')
+        value = cstr(value, 'value')
+        cdef:
+            char *_key = key
+            char *_value = value
+        with nogil:
+            ret = rbd_metadata_set(self.image, _key, _value)
+
+        if ret != 0:
+            raise make_ex(ret, 'error setting config %s for image %s' %
+                          (key, self.name))
+
+
+    def config_image_get(self, key):
+        """
+        Get an image-level configuration override.
+
+        :param key: config key
+        :type key: str
+        :returns: str - config value
+        """
+        key = 'conf_' + key
+        key = cstr(key, 'key')
+        cdef:
+            char *_key = key
+            size_t size = 4096
+            char *value = NULL
+            int ret
+        try:
+            while True:
+                value = <char *>realloc_chk(value, size)
+                with nogil:
+                    ret = rbd_metadata_get(self.image, _key, value, &size)
+                if ret != -errno.ERANGE:
+                    break
+            if ret == -errno.ENOENT:
+                raise KeyError('no config %s for image %s' % (key, self.name))
```

Comment:
Nit: this will return `conf_<key>` -- might want to use a new `conf_key` variable instead of overwriting the user provided key.

Reply:
I will use conf_key, thanks a lot.